File: JandIBogam/backend/src/main/java/com/ssafy/mvc/model/service/MedicationService.java
package com.ssafy.mvc.model.service;

import com.ssafy.mvc.model.dto.MedicationDto;

import java.util.List;

public interface MedicationService {

    MedicationDto createMedication(MedicationDto medication);
    MedicationDto getMedicationById(int id);
    List<MedicationDto> getAllMedications();
    List<MedicationDto> getMedicationsByUserId(int userId);
    MedicationDto updateMedication(MedicationDto medication);
    boolean deleteMedication(int id);
}

File: JandIBogam/frontend/JandiBogam/src/services/MedicationService.js
// src/services/MedicationService.js
import apiClient from './api'

const MedicationService = {
  createMedication(medicationData) {
    return apiClient.post('/medications', medicationData)
  },

  getMedicationById(id) {
    return apiClient.get(`/medications/${id}`)
  },

  getMedicationsByUserId(userId) {
    return apiClient.get(`/medications/users/${userId}`)
  },

  updateMedication(id, medicationData) {
    return apiClient.put(`/medications/${id}`, medicationData)
  },

  deleteMedication(id) {
    return apiClient.delete(`/medications/${id}`)
  }
}

export default MedicationService
